presto-main-base/src/main/java/com/facebook/presto/execution/scheduler/nodeSelection/SimpleNodeSelector.java

```diff
@@ -149,6 +158,11 @@ public SplitPlacementResult computeAssignments(Set<Split> splits, List<RemoteTas
         Set<InternalNode> blockedExactNodes = new HashSet<>();
         boolean splitWaitingForAnyNode = false;
 
+        ToLongFunction<InternalNode> taskLoadSplitWeightProvider = null;
```

Comment:
see suggestion below to move this down

Reply:
@rschlussel 
Hmm, I don't see any suggestions to move this down...
It should be before the 'for' loop.
We could move it close to the "NodeProvider nodeProvider" initialization?

Reply:
oh i see.  then I recommend making this an optional so it's clear that it might not be set (and then later check if it's present rather than not null). But otherwise this is okay.